### ISSUE TITLE
All of the GKE Upgrade tests for 1.2 -> 1.3 are failing because of an issue in teardown

### ISSUE THREAD

**Issue by maisem** [kubernetes/test-infra#539]
All of the GKE Upgrade tests for 1.2 -> 1.3 are failing because of an issue in teardown.
The tests never actually run.

```
go run /workspace/e2e.go --check_version_skew=false -v --dump=/workspace/_artifacts --check_leaked_resources --up '--ctl=version --match-server-version=false' --down --test '--test_args= --kubectl-path=/workspace/kubernetes_skew/cluster/kubectl.sh' '--upgrade_args=--ginkgo.focus=\[Feature:ClusterUpgrade\] --upgrade-target=ci/latest-1.3'
2016/09/08 19:24:07 e2e.go:449: Running: teardown
... in gke:prepare-e2e()
... in gke:test-teardown()
... in gke:detect-project()
ERROR: (gcloud.compute.firewall-rules.delete) Some requests did not succeed:
 - The resource 'projects/kube-gke-upg-1-2-1-3-upg-clu/global/firewalls/jenkins-e2e-http-alt' was not found

ERROR: (gcloud.compute.firewall-rules.delete) Some requests did not succeed:
 - The resource 'projects/kube-gke-upg-1-2-1-3-upg-clu/global/firewalls/jenkins-e2e-nodeports' was not found

Bringing down cluster using provider: gke
```

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.2-1.3-upgrade-cluster/4076
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.2-1.3-upgrade-cluster-new/4517
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.2-1.3-upgrade-master/4057
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.2-1.3-kubectl-skew/6080

cc: @spxtr @ixdy @fabioy @vishh @pwittrock

**Comment by maisem** [comment]
Make that all 1.2 -> 1.3/1.4 tests are failing because of the same issue.
Also looks like we are missing 1.2->1.4 kubectl skew tests.

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.2-1.4-upgrade-cluster/5844
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.2-1.4-upgrade-cluster-new/5871
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.2-1.4-upgrade-master/5868

**Comment by fejta** [comment]
@ixdy is this another instance of the famed cannot parse gcloud output problem? If so will you pull that into 1.2? Thanks!

**Comment by fejta** [comment]
This actually looks like an incompatibility between 1.2 and master in the behavior in test-teardown:

https://github.com/kubernetes/kubernetes/blob/release-1.2/cluster/gke/util.sh#L301

https://github.com/kubernetes/kubernetes/blob/f236df504a5944eb227a0d97dde63bed9667e9dd/cluster/gke/util.sh#L366

**Comment by ixdy** [comment]
https://github.com/kubernetes/kubernetes/pull/30942 has been sitting for a while. It's not going to auto-merge, because I think node e2e tests are pretty broken on release-1.2.

**Comment by maisem** [comment]
@ixdy - kubernetes/kubernetes#30942 has been merged by @roberthbailey.

**Comment by maisem** [comment]
Still not working.

**Comment by davidopp** [comment]
ping @ixdy

**Comment by ixdy** [comment]
@fejta was just talking to me about the teardown issues on 1.2. I'm not sure what the plan is.

**Comment by ixdy** [comment]
The issue is that `hack/e2e.go` now tries to tear down any existing cluster before starting a cluster, regardless of whether a cluster already exists.

In 1.2, tearing down a nonexistent cluster failed. In 1.3+, this operation passes.

We could possibly revert the change in e2e.go so that we only try to teardown when the cluster is up.

**Comment by ixdy** [comment]
https://github.com/kubernetes/kubernetes/pull/32420 for your consideration.

**Comment by fejta** [comment]
Thanks @ixdy!

**Comment by maisem** [comment]
Thanks @ixdy!

**Comment by ixdy** [comment]
My previous fix broke the GKE build because (conveniently) shortly after it was activated, a GKE cluster failed to start up in time ([log](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke/15244)), resulting in all subsequent runs to fail (https://github.com/kubernetes/kubernetes/issues/32431) because the cluster already existed ([example log](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke/15245)).

I expected the `IsUp()` guard to detect that the cluster was running, and thus tear it down, but `IsUp()` doesn't work properly in an empty Jenkins workspace with no `~/kube/config`, I think:
```
2016/09/10 20:34:34 e2e.go:439: Running: get status
... in gke:prepare-e2e()
Client Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.0-alpha.0.515+b73aa599017a57", GitCommit:"b73aa599017a573ed54b36304202734c5248f894", GitTreeState:"clean", BuildDate:"2016-09-11T03:23:43Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
The connection to the server localhost:8080 was refused - did you specify the right host or port?
2016/09/10 20:34:34 e2e.go:441: Step 'get status' finished in 91.411293ms
```

I'm not sure if this check ever worked correctly.

We're reverting my change in https://github.com/kubernetes/kubernetes/pull/32521, and then @rmmh will backport a fix to `release-1.2` so that we can do test teardown every time, not failing if the cluster isn't up.

(I'm not sure if we need to cut a new 1.2 release for that fix to work. I hope not.)
